File: jpa-backend/src/main/java/chaincue/tech/jpabackend/masters/course_master/Course.java
package chaincue.tech.jpabackend.masters.course_master;

import chaincue.tech.jpabackend.masters.student_master.Student;
import chaincue.tech.jpabackend.masters.teacher_master.Teacher;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToMany;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.List;
import java.util.UUID;

@Entity(name = "courses")
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString(exclude = {"teacherCourseRelations", "studentCourseRelations"})
public class Course {
    @Id
    String id;
    String name;
    @ManyToMany(mappedBy = "teacherCourseRelations")
    List<Teacher> teacherCourseRelations;
    @ManyToMany(mappedBy = "studentCourseRelations")
    List<Student> studentCourseRelations;

    public static Course createCourse(String name) {
        return new Course(
                UUID.randomUUID().toString(),
                name,
                List.of(),
                List.of()
        );
    }
}

File: kotlin/jpa-backend/src/main/kotlin/chaincue/tech/jpabackend/masters/course_master/Course.kt
package chaincue.tech.jpabackend.masters.teacher_master

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.ManyToMany
import lombok.ToString
import java.util.*

@ToString(exclude = ["teacherCourseRelations", "studentCourseRelations"])
@Entity(name = "courses")
data class Course(
        @Id
        val id: String,
        val name: String,
        @ManyToMany(mappedBy = "teacherCourseRelations")
        val teacherCourseRelations: MutableList<Teacher>,
//        @ManyToMany(mappedBy = "studentCourseRelations")
//        val studentCourseRelations: MutableList<Student>

) {
    companion object {
        fun create(username: String): Course = Course(
                id = UUID.randomUUID().toString(),
                name = username,
                teacherCourseRelations = mutableListOf(),
//                studentCourseRelations = mutableListOf()
        )
    }

    override fun toString(): String {
        return "Course(id='$id', name='$name')"
    }

}
